07-routing-gossip.md

```diff
@@ -798,6 +838,33 @@ and allows bootstrapping for new nodes as well as updating for nodes that
 have been offline for some time.  The `gossip_queries` option
 allows for more refined synchronization.
 
+### Inventory-based Rebroadcasting
+
+With the simple rebroadcasting scheme described above, nodes that are connected to many different peers will receive duplicates for the same gossip messages.
+
+A more efficient scheme based on inventory messages, similar to how transactions and blocks are propagated on the bitcoin p2p network, can optionally be used: if the `option_inv_gossip` feature bit is set, instead of broadcasting `channel_annoucement` and `channel_update` messages, nodes will broadcast a `gossip_inventory` message which is similar to a `reply_channel_range` message and can be used to filter announcements by timestamp or by content (checksum):
+
+1. type: 266 (`gossip_inventory`)
```

Comment:
Rather than a single message for all inventory types I think we should instead have distinct messages as this allows us to open encode certain fields that are relevant if a node wants to filter out a class of advertisements. For example a `channel_inv` message would have things like the size of the channel, or the nodes that it's connecting included. This would let nodes performing filtering such as _ignoring_ all channels that are below a certain size. With the checksum approach the node has no idea what it's about to fetch, and will waste bandwidth by fetching something that it will instantly abandon. 

While we're at it, we may also need to extend the prior queries in order to allow fetching a node announcement.

Reply:
I've rebased to get the latest changes from #557, and I'm using TLV encoding now. I've adding a node inventory messages (just a list of node ids). 
I don't think that we need to include channel capacity in the inv message: you would have to trust the sender anyway, and tiny channels will progressively go away.
I don't think we should include the node ids it's connecting: how would you use it ? You don't know what a new channel enables from a routing pov until you have the rest of the graph